source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_metadata_file_reader.cc

```diff
@@ -0,0 +1,54 @@
+#include "source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_metadata_file_reader.h"
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "envoy/common/exception.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+namespace {
+
+bool isIndirectionFile(const std::string& file_name) {
+  return absl::StartsWith(file_name, "dt_metadata_");
+}
+
+std::string readFile(const std::string& file_name) {
+  if (file_name.empty()) {
+    return "";
+  }
+
+  std::ifstream file(file_name);
+  if (file.fail()) {
+    throw EnvoyException(absl::StrCat("Unable to read Dynatrace enrichment file: ", file_name));
+  }
+
+  std::stringstream file_string;
+  file_string << file.rdbuf();
+
+  return file_string.str();
+}
+
+} // namespace
+
+std::string DynatraceMetadataFileReaderImpl::readEnrichmentFile(const std::string& file_name) {
+  if (const bool indirectionFile = isIndirectionFile(file_name); indirectionFile) {
```

Comment:
nit: `indirection_file`. :)

Reply:
Sorry, forgot about that one. Fixed in [a4f9bb0](https://github.com/envoyproxy/envoy/pull/30824/commits/a4f9bb03436540b46c90c977e3840f9715bdbf5a).